job/cache.go

```diff
@@ -115,6 +115,18 @@ func (c *MemoryJobCache) Delete(id string) {
 	c.jobs.Lock.Lock()
 	defer c.jobs.Lock.Unlock()
 
+	j := c.jobs.Jobs[id]
+	if j == nil {
+		// TODO
+		//return nil, JobDoesntExistErr
+		return
+	}
+
+	// TODO return error
+	c.jobs.Lock.Unlock()
```

Comment:
I see now... why the extra unlocking and locking? Is there some reason why the line beneath this comment, 127, should not be locked?